### PR TITLE
[Fix #337][Fix #345][Fix #341] Static Site build scripts

### LINKED ISSUE #337: Strip down requirements.txt to the absolute minimum required to build the static site

### LINKED ISSUE #341: Include robots.txt and crontibute.json to the static site dump

### LINKED ISSUE #345: Create script to generate a static version of the site

### PR DESCRIPTION
**Pull request by glogiotatidis**
- ~~Fix links to listing page with filtering~~.  moved to #361

### CODE REVIEW

**Comment by glogiotatidis** [comment]
Marking this one WIP waiting for tomorrow's discussion on how we will automate it. 

/cc @mozmeao/theconn

**Comment by jgmize** [comment]
This PR removes our security header implementations as well, which reminds me that we will need to re-implement them for a static site, which isn't possible to do with s3 (or GCS) alone, but can be done if we add [Cloudflare](https://scotthelme.co.uk/security-headers-cloudflare-worker/) or [Cloudfront](https://medium.com/@tom.cook/edge-lambda-cloudfront-custom-headers-3d134a2c18a2) (but not [Google CDN](https://issuetracker.google.com/issues/36427250)).

**Comment by pmac** [comment]
It's easy with [Netlify](https://www.netlify.com/docs/headers-and-basic-auth/)... just FYI :)

**Comment by glogiotatidis** [comment]
> This PR removes our security header implementations as well, which reminds me that we will need to re-implement them for a static site, which isn't possible to do with s3 (or GCS) alone, but can be done if we add [Cloudflare](https://scotthelme.co.uk/security-headers-cloudflare-worker/) or [Cloudfront](https://medium.com/@tom.cook/edge-lambda-cloudfront-custom-headers-3d134a2c18a2) (but not [Google CDN](https://issuetracker.google.com/issues/36427250)).

Added #362 to track this

**Comment by glogiotatidis** [comment]
> It's easy with [Netlify](https://www.netlify.com/docs/headers-and-basic-auth/)... just FYI :)

Would like to try Netlify for this.

**Comment by glogiotatidis** [comment]
I created a new branch named `static` and changed it to be the target for this PR, so we can co-ordinate and fix the [Milestone Issues](https://github.com/mozmeao/lumbergh/milestone/1) without yet touching the Django production site. Added a few more issues based on what we discussed yesterday

This should be ready for someone to review, as a mostly generic approach of building a static version of the site. I would probably need to adjust `build.sh` when we decide on build method.

File: docker-compose-build.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  web:
+    build: .
+    volumes:
+      - ./website_copy:/tmp/careers.mozilla.org/
+    environment:
+      - PYTHONDONTWRITEBYTECODE=1
+      - DATABASE_URL=sqlite:///database.sqlite
+      - DEBUG=False
+      - ALLOWED_HOSTS=*
+      - SECRET_KEY=59114b6a-2858-4caf-8878-482a24ee9542
```
[the remark this answers is not in the document]
Comment (glogiotatidis) [reply docker-compose-build.yml]: Yes these are fine. Thanks for checking!

File: bin/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+# Create directory and ensure it's empty
+mkdir -pv website_copy
+rm -rf website_copy/*
+
+# Build static copy
+docker-compose -f docker-compose-build.yml -p careers-${BRANCH_NAME:-dev}-${BUILD_NUMBER:-1} run web ./bin/mirror.sh
+
+# Sync to S3 bucket
+pushd website_copy
```
[the remark this answers is not in the document]
Comment (glogiotatidis) [reply bin/build.sh]: Not really, will fix

**Comment by glogiotatidis** [comment]
OK This is updated to work with netlify. You can check the PR deployed at https://deploy-preview-359--careers-stage.netlify.com/

I has been very easy to run the `mirror.sh` script on netlify and I overall very much liked the workflow, the checks they run and the build speed.

I'm +1 for running this on netlify. We have a few things to take care after merging this like (#360 #361 #362)

@mozmeao/theconn thoughts / reviews?

**Comment by metadave** [comment]
This looks great! Clicking around the `deploy/netlify` check above, the site appears to be working correctly. Are there any tests that would detect broken links in the rendered HTML?

Next steps would be to get security and billing squared away w/ Netlify.

File: bin/mirror.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+# Wait for site to come online
+CHECK_PORT=8000 CHECK_HOST=0.0.0.0 ./bin/takis
+
+mkdir -vp _site
+pushd _site
+
+wget --header "Host: ${CAREERS_HOST:-careers.mozilla.org}" \
+     --reject-regex "(.*)\?(.*)|static/(.*)" \
+     --mirror \
+     -nH -p -E http://0.0.0.0:8000/ http://0.0.0.0:8000/contribute.json
```
Comment (jpetto) [inline bin/mirror.sh]: Could use a comment on this command. Lots of flags used that I definitely don't have memorized. 😉

File: netlify-headers [hunk 4]
```diff
@@ -0,0 +1,8 @@
+/*
+  X-Frame-Options: DENY
+  X-XSS-Protection: 1; mode=block
+  X-Content-Type-Options: nosniff
+  Content-Security-Policy: font-src 'self' http://*.mozilla.net https://*.mozilla.net http://*.mozilla.org https://*.mozilla.org; img-src 'self' data: http://*.mozilla.net https://*.mozilla.net http://*.mozilla.org https://*.mozilla.org http://www.google-analytics.com https://www.google-analytics.com; style-src 'self' 'unsafe-inline' http://*.mozilla.org https://*.mozilla.org http://*.mozilla.net https://*.mozilla.net; script-src 'self' data: http://*.mozilla.org https://*.mozilla.org http://*.mozilla.net https://*.mozilla.net http://www.google-analytics.com https://www.google-analytics.com http://www.googletagmanager.com https://www.googletagmanager.com; default-src 'self' *.cdn.mozilla.net assets.mozilla.net
```
Comment (jpetto) [inline netlify-headers]: Not a big deal, but would be nice if this list could be stored in a more readable format.

**Comment by jpetto** [comment]
FWIW, here are the WebPageTest and Observatory results for current vs Netlify:

(Spoiler alert - they're basically identical, with Netlify winning in performance.)

## Current
- [WebPageTest](https://www.webpagetest.org/result/190123_WW_67551acb0ac632624ccbf77eaaa4ff4c/)
- [Observatory](https://observatory.mozilla.org/analyze/careers.mozilla.org)

## Netlify
- [WebPageTest](https://www.webpagetest.org/result/190123_GP_3f6054f7909a4f0c319a9810cba56037/)
- [Observatory](https://observatory.mozilla.org/analyze/deploy-preview-359--careers-stage.netlify.com)

**Comment by jpetto** [comment]
Oh, I also ran the Netlify site through a broken link checker and it found none. (Well, fine, it found one, but it was in a job posting, which doesn't count/matter here.)

File: netlify-headers (hunk 4 above)
Comment (glogiotatidis) [reply netlify-headers]: Agreed but that's the way netlify wants that.

File: bin/mirror.sh (hunk 3 above)
Comment (glogiotatidis) [reply bin/mirror.sh]: Added!

**Comment by glogiotatidis** [comment]
Added wget docs and an additional acceptance test. I'm going to merge this one to `static` branch and continue to work against this branch until we're ready to go live with this.

Thanks for your time all!

File: netlify-headers (hunk 4 above)
Comment (pmac) [reply netlify-headers]: We could generate this file, but that is more work and this stuff is unlikely to change very often.

**Pull request merged**